src/pathDetails/PathDetails.tsx

```diff
@@ -0,0 +1,184 @@
+            //       them just for this...
+            expandControls: false,
+        }
+        const callbacks = {
+            pointSelectedCallback: onPathDetailHover,
+            areaSelectedCallback: onRangeSelected,
+            routeSegmentsSelectedCallback: onElevationSelected,
+        }
+        setGraph(new HeightGraph(containerRef.current, options, callbacks))
+    }, [containerRef])
+    const resizeGraph = () => {
+        graph?.resize({ width: containerRef.current?.clientWidth, height: containerRef.current?.clientHeight })
+    }
+    useEffect(() => {
+        window.addEventListener('resize', resizeGraph)
+        return () => window.removeEventListener('resize', resizeGraph)
+    })
+    useEffect(() => {
+        const pathDetailsData = buildPathDetailsData(selectedPath)
+        graph?.setData(pathDetailsData.data, pathDetailsData.mappings)
+    }, [selectedPath, graph])
+    return (
+        <div className={styles.layoutContainer}>
+            <div className={styles.heightgraphContainer} ref={containerRef} />
+        </div>
+    )
+}
+
+/** executed when we hover the mouse over the path details diagram */
+function onPathDetailHover(point: Coordinate, elevation: string, description: string) {
+    Dispatcher.dispatch(new PathDetailsHover(point ? { point, elevation, description } : null))
+}
+
+/** executed when we box-select a range of the path details diagram */
+function onRangeSelected(bbox: { sw: Coordinate; ne: Coordinate } | null) {
+    // bbox = null means that the range was cleared
+    Dispatcher.dispatch(
+        new PathDetailsRangeSelected(bbox ? [bbox.sw.lng, bbox.sw.lat, bbox.ne.lng, bbox.ne.lat] : null)
+    )
+}
+
+/** executed when we use the vertical elevation slider on the right side of the diagram */
+function onElevationSelected(segments: Coordinate[][]) {
+    Dispatcher.dispatch(new PathDetailsElevationSelected(segments))
+}
+
+function buildPathDetailsData(selectedPath: Path) {
+    const elevation = createFeatureCollection('Elevation [m]', [
+        createFeature(selectedPath.points.coordinates, 'elevation'),
+    ])
+    const pathDetails = Object.entries(selectedPath.details).map(([detailName, details]) => {
+        const points = selectedPath.points.coordinates
+        const features = details.map(([from, to, value = 'Undefined']: [number, number, string | number]) =>
+            createFeature(points.slice(from, to + 1), value)
+        )
+        return createFeatureCollection(detailName, features)
+    })
+    const mappings: any = {
+        'Elevation [m]': function () {
+            return { text: 'Elevation [m]', color: '#27ce49' }
+        },
+    }
+    Object.entries(selectedPath.details).forEach(([detailName, details]) => {
+        mappings[detailName] = createColorMapping(details)
+    })
+    return {
+        data: [elevation, ...pathDetails],
+        mappings,
+    }
+}
+
+function createColorMapping(detail: any): any {
```

Comment:
I know you've ported it from js, but maybe remove the anys? Probably future you will be happy about it.

Reply:
I added some types here even though I am not fully convinced this improved much :) But generally I agree that adding most or all the types will save time down the road. Here it is a bit awkward because we get these path details from the GH API which could either be `[number, number, number][]` or `[number, number, string][]` and there already is some code that needs to distinguish the two cases.